client/v3/internal/endpoint/endpoint.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"net/url"
+	"regexp"
+)
+
+var (
+	STRIP_PORT_REGEXP = regexp.MustCompile("(.*):([0-9]+)")
+)
+
+func stripPort(ep string) string {
+	return STRIP_PORT_REGEXP.ReplaceAllString(ep, "$1")
+}
+
+func translateEndpoint(ep string) (addr string, serverName string, requireCreds bool) {
+	url, err := url.Parse(ep)
+	if err != nil {
+		return ep, stripPort(ep), false
```

Comment:
The return value is named `serverName`, and we are using `stripPort`... Can we change the return name or comment?

Reply:
Fixed in https://github.com/etcd-io/etcd/pull/12706